src/styles/getMuiTheme.js

```diff
@@ -240,7 +242,9 @@ export default function getMuiTheme(baseTheme, muiTheme) {
     },
   }, muiTheme);
 
+  const transformers = [autoprefixer, rtl, callOnce].map(t => t(muiTheme)).filter(t => t);
```

Comment:
That's already happening, remember we change the implementation to do the production check within the transformer transformers/callOnce.js#L6. 

If that method returns undefined, then it is not included in the transfomer chain.  The `filter` method on this line of code is filtering all transformer "creators" that are returning a falsy value (such as when the environment is in production).